lib/std/private/win_setenv.nim

```diff
@@ -30,44 +30,46 @@ else:
     # same as winlean.setEnvironmentVariableA
 
   proc c_getenv(env: cstring): cstring {.importc: "getenv", header: "<stdlib.h>".}
-  proc c_putenv_s(envname: cstring, envval: cstring): cint {.importc: "_putenv_s", header: "<stdlib.h>".}
+  proc c_putenv(envstring: cstring): cint {.importc: "_putenv", header: "<stdlib.h>".}
   proc c_wgetenv(varname: ptr wchar_t): ptr wchar_t {.importc: "_wgetenv", header: "<stdlib.h>".}
 
   var errno {.importc, header: "<errno.h>".}: cint
-  var gWenviron {.importc:"_wenviron".}: ptr ptr wchar_t
+  var gWenviron {.importc: "_wenviron".}: ptr ptr wchar_t
     # xxx `ptr UncheckedArray[WideCString]` did not work
 
-  proc mbstowcs_s(pReturnValue: ptr csize_t, wcstr: ptr wchar_t, sizeInWords: csize_t, mbstr: cstring, count: csize_t): cint {.importc: "mbstowcs_s", header: "<stdlib.h>".}
+  proc mbstowcs(wcstr: ptr wchar_t, mbstr: cstring, count: csize_t): csize_t {.importc: "mbstowcs", header: "<stdlib.h>".}
     # xxx cint vs errno_t?
 
-  proc setEnvImpl*(name: cstring, value: cstring, overwrite: cint): cint =
+  proc setEnvImpl*(name: string, value: string, overwrite: cint): cint =
     const EINVAL = cint(22)
-    const MAX_ENV = 32767
-      # xxx get it from: `var MAX_ENV {.importc: "_MAX_ENV", header:"<stdlib.h>".}: cint`
-    if overwrite == 0 and c_getenv(name) != nil: return 0
-    if value[0] != '\0':
-      let e = c_putenv_s(name, value)
+    let cname = cstring(name)
+    if overwrite == 0 and c_getenv(cname) != nil: return 0
+    if value != "":
+      let cenvstring = cstring(name & "=" & value)
```

Comment:
This pattern is bad. With --gc:orc the temporary string is freed "immediately" (at the scope exit, but still). Instead make the `let` the owner of the expression (aka keep it as `string`) and convert it to `cstring` at the callsites.